format/spec.md

```diff
@@ -113,9 +122,9 @@ Tables do not require random-access writes. Once written, data and metadata file
 Tables do not require rename, except for tables that use atomic rename to implement the commit operation for new metadata files.
 
 
-## Specification
+# Specification
 
-#### Terms
+### Terms
```

Comment:
so style here is to skip "##" to avoid a table of contents entry?  I wonder if it is worth a comment?

Reply:
I was trying to avoid this being at the same level as things like `Schemas and data types` further down. These aren't major sections of the spec. That's why I thought h3 rather than h2 was the right choice. I think we'll probably need to adjust these when we see what the site generates for the TOC.

Reply:
OK, I don't have a strong opinion, but we might want to leave a comment if possible so future readers don't try to correct what they might see as a mistake?

Reply:
TOC is not generated now for spec and its sub section. It only covers first H1 and its sub topics. 
<img width="316" alt="image" src="https://github.com/user-attachments/assets/549daee4-477f-42dc-bc21-bf14af3d230e">


Without the table of contents for spec and its sub topics it is hard to navigate. 
I have added a detailed description and fixed the same in https://github.com/apache/iceberg/pull/11067